### PR TITLE
Improve and unify names

### PR DESCRIPTION
**Pull request by ssiccha**
RecognitionInfoFamily -> RecogNodeFamily
IsRecognitionInfo -> IsRecogNode
RIFac -> ImageRecogNode
RIKer -> KernelRecogNode

PR 1 for #177

### CODE REVIEW

**Comment by ssiccha** [comment]
Merging this may clash with #176. I'll try whether I can rebase that branch easily on top of this one. Otherwise we'd have to wait until #176 is merged first.

File: doc/recognition.xml
Comment (fingolfin) [inline doc/recognition.xml:210]: I don't like that that these names do not reference the fact that this is for recognition. If I here "image node" then I don't know what this is about at all.

Elsewhere in this PR you introduce  `IsRecogNode`. So how about calling these `ImageRecogNode` or `RecogNodeForImage` or so (and similar for the kernel)?

Comment (ssiccha) [reply doc/recognition.xml:210]: I like `ImageRecogNode`. 👍

**Comment by ssiccha** [comment]
@FriedrichRober The tests fail because we rename e.g. `RIFac` but not `Has... ` and `SetRIFac`. We need to update our script accordingly.

File: doc/renaming_table.xml
Comment (wilfwilson) [inline doc/renaming_table.xml:30]: Why is there a closing `Chapter` tag here?

File: doc/renaming_table.xml
Comment (wilfwilson) [inline doc/renaming_table.xml:1]: Do you intend to include `doc/renaming_table/xml` in the manual somewhere?

**Review by wilfwilson — COMMENTED** [review]
In general I approve, while also supporting Max's suggestion.

This is perhaps an ignorant question (I've not really thought about recog for quite a while) - is naming this in terms of 'image' definitely more appropriate than naming it in terms of 'factor'? I only ask because (I presume) the old name `RIFac` was referring to a 'factor'; although the actual documentation talked about the 'image'.

File: doc/renaming_table.xml
Comment (ssiccha) [reply doc/renaming_table.xml:1]: Yepp. I haven't looked into the renaming code for ages and I forgot to add a few changes. :grin:

**Comment by wilfwilson** [comment]
Cool 🙂 I'm happy for it to be called `image`, but as just curious.

**Comment by ssiccha** [comment]
Rebased onto master, fixed the tests and the documentation. Please in particular have a look at the file `doc/renaming.xml` or Chapter 8 "Renaming".

File: doc/renaming_table.xml
Comment (wilfwilson) [inline doc/renaming_table.xml:10]:
```suggestion
	<Item> <C>RecognitionInfoFamily</C> </Item>
```
and so on.

Also, perhaps index keys could be added for `RecognitionInfoFamily`, etc, so that the old names still appear in the index (but they now direct you to the correct place now, so `RecognitionInfoFamily` would take you to the documentation for `RecogNodeFamily`, etc). I _think_ this is popular with GAP Doc, but I can't remember how.

Comment (wilfwilson) [reply doc/renaming_table.xml:10]: Perhaps you have to add `<Index><C>RecognitionInfoFamily</C></Index>` to the documentation for `RecogNodeFamily`.

Comment (ssiccha) [reply doc/renaming_table.xml:10]: That works for me. Nice, thanks! Adding the old names to the index sounds like a good idea to me.

I think I would prefer it to have the index entries of the old names link to the renaming chapter instead. That way there can't be any confusion about why one clicks e.g. on `RecognitionInfoFamily`, but does not find any documentation for `RecognitionInfoFamily`. Especially since sometimes a new name @FriedrichRober and I came up with can be very different from the old name, if the latter was very cryptic.

Comment (wilfwilson) [reply doc/renaming_table.xml:10]: Yeah that's a better idea.

Comment (ssiccha) [reply doc/renaming_table.xml:10]: In order for such an entry to be sorted correctly into the index one also needs to specify a sort `Key` like this:
`<Index Key="RecognitionInfoFamily"><C>RecognitionInfoFamily</C></Index>`

**Review by FriedrichRober — APPROVED** [review]
I think the names are now very descriptive and the renamings are well documented.

**Comment by fingolfin** [comment]
Regarding factor vs. image: of course both make sense. However, we do talk about the "factor group" in many comments and  the manual. There also still the attributes `methodsforfactor` or `forfactor`, which now don't match anymore. Of course w could also rename them.

Taking a step back, I guess both names make sense: a factor group is the image of a group homomorphism... However, my mind has a slight tendency to think of this as a "proper factor", i.e. the homomorphism has a non-trivial kernel, while in our case, there are also sometimes isomorphisms... While "image" does not have such connotation to me. So maybe that means "image" would be better indeed, at least if other people suffer from the same prejudice regarding "factor" as I do :-)

**Comment by ssiccha** [comment]
Hm, interesting. When you think of an inner node in the recog tree as a homomorphism it is IMO more direct to talk of kernels and images. When you think of an inner node as a group, it is IMO more direct to talk of normal subgroups and factors. At the moment though we're mixing both by talking about kernels and factors.

And yes, when I hear or read "factor" I also don't think of a factor by the trivial subgroup. :D

**Comment by ssiccha** [comment]
Two follow up PRs would be:
- turn `Info` into `Node` where sensible
- turn `Factor` into `Image` where sensible

In both cases we should also search the manual for left over occurences of `Info` and `Factor`. I can use my tags file to search for all names which contain `Info` or `Factor`.

**Comment by wilfwilson** [comment]
I've enjoyed this discussion 🙂

**Comment by ssiccha** [comment]
I've rebased this onto master and squashed the "fixup" commits. No other changes were introduced.

**Comment by ssiccha** [comment]
I'll try to get #176 into a state where it can be merged into master. Then I'd merge in this PR (#253) and then @FriedrichRober and I can continue work on #176 in a different branch and PR.

**Comment by ssiccha** [comment]
Rebased onto master. Now I need to change `factor` in the manual to `image` and make sure that updating the index is also done in the script @FriedrichRober and I wrote.

**Comment by ssiccha** [comment]
I've replaced most occurences of `factor` with `image`. Of course, where `factor` meant a prime factor or the factor of a polynomial I left it alone. :)

**Comment by ssiccha** [comment]
@fingolfin Could you have another look?

**Comment by ssiccha** [comment]
TODO: I just saw, that `InstallMethod` calls often have a string `for recognition infos` or something similar. I need to adjust this.

**Comment by ssiccha** [comment]
Ok, I got rid of almost all appearances of `'\<info\>'`. The remaining ones don't have anything to do with recog info records.

**Review by wilfwilson — APPROVED** [review]
LGTM.

**Comment by ssiccha** [comment]
`bugfix.tst` failed because now line breaks are different.

**Pull request merged**
